### ISSUE TITLE
Use web3j.ethNewFilter() and web3j.ethGetFilterLogs() to improve ERC20 retrieval performance

### ISSUE THREAD

**Issue by medvedev1088** [xiaoyao1991/presto-ethereum#41]
Using log filters instead of retrieving transaction receipts significantly improves performance of ERC20 retrieval. 

My tests show 100x improvement for some blocks. Here is an example implementation in python https://github.com/medvedev1088/ethereum-etl/blob/master/export_erc20_transfers.py

**Comment by xiaoyao1991** [comment]
Good point. I didn't choose to implement that way because Infura does not support this API call yet.

**Comment by medvedev1088** [comment]
What do you think about ERC20RetrievalStrategy with 2 implementations ReceiptRetrievalStrategy, LogFilterRetrievalStrategy. It can be configured with a property,  erc20.retrieval-strategy=receipt|log-filter|auto.

BTW, do you know if Infura have plans to add support for filter APIs?

**Comment by xiaoyao1991** [comment]
I like that. 

I don't know if Infura will support filter APIs. It's kinda hard cuz having filters makes their API stateful.
